### PR TITLE
Able to support passing config from arguments.

### PR DESCRIPTION
**Pull request by evshary**
Able to provide configuration in the arguments to increase the flexibility.

### CODE REVIEW

File: examples/common/mod.rs [hunk 1]
```diff
@@ -10,20 +10,72 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
+use clap::Parser;
 use up_transport_zenoh::zenoh_config;
 
+#[derive(clap::ValueEnum, Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub enum Wai {
```
Comment (PLeVasseur) [inline examples/common/mod.rs:17]: `Wai` => `WhatAmI`, right? I can say for certain I would not have understand that if not for having worked with Zenoh a little. Maybe something more descriptive or leave a comment?

File: examples/common/mod.rs [hunk 2]
```diff
@@ -10,20 +10,72 @@
...
+    /// Disable the multicast-based scouting mechanism.
+    no_multicast_scouting: bool,
+}
+
 #[allow(clippy::must_use_candidate, clippy::missing_panics_doc)]
 pub fn get_zenoh_config() -> zenoh_config::Config {
+    let args = Args::parse();
+
     // Load the config from file path
-    // Config Examples: https://github.com/eclipse-zenoh/zenoh/blob/0.10.1-rc/DEFAULT_CONFIG.json5
```
Comment (PLeVasseur) [inline examples/common/mod.rs:18]: I like providing an example config file. Maybe we can add one to this repo instead of pointing to an external one?

**Review by PLeVasseur — APPROVED** [review]
Looks good. Consider adding an example config file.

File: examples/common/mod.rs (hunk 1 above)
Comment (evshary) [reply examples/common/mod.rs:17]: Agree with you. The reason I used `Wai` is to match the usage in Zenoh examples, but it's unclear as you said.

File: examples/common/mod.rs (hunk 2 above)
Comment (evshary) [reply examples/common/mod.rs:18]: Yes, I've added the config file.

**Pull request merged**